### DISCUSSION TITLE
ISSUE: Upgrade to laravel/framework 10.19.0 is breaking component: The given child should implement Spatie\Html\HtmlElement or be a string

### DISCUSSION THREAD

**Discussion by hemmesdev** [spatie/laravel-html#193] [General]
I've recently update to laravel/framework 10.19.0 and noticed that it breaks on of my components. 

Error message is: The given child should implement `Spatie\Html\HtmlElement` or be a string

My component looks like this: 
```
<!-- Send button -->
<div class="row form-group text-center">
    <div class="col-lg-12">
        {{
            html()
                ->submit($slot)
                ->class(['btn', 'btn-primary'] + (($disabled ?? false) ? ['disabled'] : []))
                ->class($class ?? [])
                ->id($id ?? 'submit')
                ->attribute(isset($form) ? 'form' : '', $form ?? '')
                ->disabled(($disabled ?? false) ? "disabled": "")
        }}
    </div>
</div>
```
And i use it like this:
```
@component('fc::submit')
       {{ __('save') }}
@endcomponent
```

I've found that the problem lies in the update to laravel/framework 10.19.0 and specifically this commit. https://github.com/laravel/framework/commit/aef975c91195a2f5a43bb9e6de09182440cf79c4

For now i have downgraded back to laravel/framework 10.18.0 and that fixes the problem, but i would be nice if a fix could be made.

**Comment by sebastiandedeyne** [comment]
Looks like there are a few places in `BaseElement` where we expect something to be a string, but we should also try to cast it to a string (since Laravel's `ComponentSlot` class implements `__toString`).

Not sure when I'll be able to look at this—PRs welcome in the mean time!

**Comment by hemmesdev** [comment]
I have created a pull with my fix to resolve the issue

#215
